### PR TITLE
chore: cleanup

### PR DESCRIPTION
**Pull request by maciejmajek**
## Purpose

Cleanup 

## Proposed Changes

Features:
- Extended import tests for all rai ROS 2 python packages
- CI now installs optional python dependencies 

Chores:
- Removed deprecated scenario_engine
- Removed communication module, moved ros_communication content to tools.utils
- Removed document module, renamed and moved loader to apps folder
- Merged tool_runner utility into rai.tools.utils
- Removed deprecated tools (cat_agri tools, mock tools etc)

## Issues

-

## Testing

- create_robots_whoami ✅ 
- streamlit gui ✅

### CODE REVIEW

File: src/rai/rai/tools/ros/native.py [hunk 1]
```diff
@@ -68,6 +68,7 @@ class PubRos2MessageToolInput(BaseModel):
 
 # --------------------- Tools ---------------------
 class Ros2BaseTool(BaseTool):
+    # TODO: Make the decision between rclpy.node.Node and RaiNode
     node: rclpy.node.Node = Field(..., exclude=True, include=False, required=True)
```
Comment (rachwalk) [inline src/rai/rai/tools/ros/native.py]:
```suggestion
    node: rclpy.node.Node = Field(..., exclude=True, required=True)
```

File: src/rai/rai/tools/utils.py [hunk 2]
```diff
@@ -91,6 +95,86 @@ def wait_for_message(
...
+            {
+                "type": "image_url",
+                "image_url": {
+                    "url": f"data:image/jpeg;base64,{image}",
+                },
+            }
+            for image in images
+        ]
+    else:
+        raise ValueError(f"Vendor {vendor} not supported")
```
Comment (rachwalk) [inline src/rai/rai/tools/utils.py:112]: TODO: Add an Issue for covering more vendors, mark `Good first issue`.

**Review by rachwalk — APPROVED** [review]

File: src/rai/rai/tools/ros/native.py [hunk 3]
```diff
@@ -68,6 +68,7 @@ class PubRos2MessageToolInput(BaseModel):
 
 # --------------------- Tools ---------------------
 class Ros2BaseTool(BaseTool):
+    # TODO: Make the decision between rclpy.node.Node and RaiNode
```
Comment (boczekbartek) [inline src/rai/rai/tools/ros/native.py:71]: FYI: using `RaiNode` here causes cicrular import error

**Review by boczekbartek — APPROVED** [review]

File: src/rai/rai/tools/ros/native.py (hunk 3 above)
Comment (maciejmajek) [reply src/rai/rai/tools/ros/native.py:71]: Yeah, had that problem. Importing under
```python
if TYPECHECKING:
 import ... RaiNode
 
# and then
node: "rclpy.node.Node | RaiNode"
 ```
 Does not work either. Nonetheless RaiNode is better suited here.

**Pull request merged**